legacy/firmware/fsm_msg_common.h

```diff
@@ -24,7 +24,8 @@ bool get_features(Features *resp) {
 #else
   const image_header *hdr =
       (const image_header *)FLASH_PTR(FLASH_FWHEADER_START);
-  if (SIG_OK == signatures_new_ok(hdr, NULL)) {
+  // only v3 signatures count as signed
+  if (SIG_OK == signatures_ok(hdr, NULL, true)) {
```

Comment:
does this mean that a production firmware will return "UNSAFE, DO NOT USE" if the installed image has a valid v2 header? that does not seem correct

Reply:
i see we already discussed this in https://github.com/trezor/trezor-firmware/issues/2513#issuecomment-1279166574

but ISTM this should be `signatures_match` anyway

Reply:
since this is in FW, at this point it should only have v3 header no?

Reply:
Not if this is a direct upgrade (no intermediate). But that is a good point. Maybe we want to always upgrade via intermediate?

Reply:
right, direct upgrade could cause that.

i think it would be a bit preferable to upgrade only via intermediate, having different headers depending on how you upgraded doesn't seem very clean

Reply:
I think the logic was that the upgrade will be only via intermediate (which does not have any USB communication), so new firmware should always be signed only by v3 signature. But the change is trivial, so it may be changed to show both v2 and v3 as valid.